evap/staff/importers/enrollment.py

```diff
@@ -0,0 +1,666 @@
+
+
+class IsGradedImportMapper:
+    class InvalidIsGradedException(Exception):
+        def __init__(self, *args, invalid_is_graded: str, **kwargs):
+            self.invalid_is_graded: str = invalid_is_graded
+            super().__init__(*args, **kwargs)
+
+    @classmethod
+    def is_graded_from_import_string(cls, is_graded: str):
+        is_graded = is_graded.strip()
+        if is_graded == settings.IMPORTER_GRADED_YES:
+            return True
+        if is_graded == settings.IMPORTER_GRADED_NO:
+            return False
+
+        raise cls.InvalidIsGradedException(invalid_is_graded=is_graded)
+
+
+@dataclass
+class EnrollmentInputRow(InputRow):
+    """Raw representation of a semantic enrollment importer row, independent on the import format (xls, csv, ...)"""
+
+    # pylint: disable=too-many-instance-attributes
+
+    column_count = 12
+
+    location: ExcelFileLocation
+
+    student_first_name: str
+    student_last_name: str
+    student_email: str
+
+    responsible_first_name: str
+    responsible_last_name: str
+    responsible_email: str
+    responsible_title: str
+
+    evaluation_name_de: str
+    evaluation_name_en: str
+    evaluation_degree_names: str
+    evaluation_course_type_name: str
+    evaluation_is_graded: str
+
+    def __init__(self, location: ExcelFileLocation, cells: Sequence[str]):
+        # pylint: disable=super-init-not-called
+        self.location = location
+
+        self.student_last_name = cells[1]
+        self.student_first_name = cells[2]
+        self.student_email = cells[3]
+
+        self.evaluation_degree_names = cells[0]
+        self.evaluation_course_type_name = cells[4]
+        self.evaluation_is_graded = cells[5]
+        self.evaluation_name_de = cells[6]
+        self.evaluation_name_en = cells[7]
+
+        self.responsible_title = cells[8]
+        self.responsible_last_name = cells[9]
+        self.responsible_first_name = cells[10]
+        self.responsible_email = cells[11]
+
+
+@dataclass
+class EnrollmentParsedRow:
+    """
+    Representation of an Enrollment Row after parsing the data into the resulting data structures.
+    For example, the course degree will already be resolved in here.  This is the data structure we want to work with.
+    """
+
+    location: ExcelFileLocation
+
+    student_data: UserData
+    responsible_data: UserData
+    course_data: CourseData
+
+
+@dataclass
+class EnrollmentPartiallyParsedRow:
+    """
+    Sometimes, we can not instantly get a CourseData from an InputRow because parsing some attributes may fail, but we
+    do not want to abort instantly. This class models a EnrollmentParsedRow where the course data may be incomplete
+    due to parsing failure.
+    """
+
+    location: ExcelFileLocation
+
+    student_data: UserData
+    responsible_data: UserData
+    partial_course_data: PartialCourseData
+
+    def get_enrollment_parsed_row(self):
```

Comment:
I personally don't like the word `get` and would prefer `as`, `into` or so here too :)